src/CustomerRepository.js

```diff
@@ -1,13 +1,23 @@
+import Customer from "./Customer";
+
 class CustomerRepository {
   constructor(allCustomerData) {
-    this.allCustomers = allCustomerData || [];
+    this.allCustomers = this.makeAllCustomers(allCustomerData) || [];
   }
 
+  makeAllCustomers(data) {
+    return data.map((data) => {
+      return new Customer(data.id, data.name)
+    })
+  }
+  
```

Comment:
Currently, if `data` is not defined it will throw an exception and won't default to `[]`. The reason because it's trying to iterate through undefined. A few ways to avoid this: 

1. If /Else Conditional in the `makeAllCustomers` method
2. Default the parameter to `[]`
3. Optional chaining on `data`
4. Use a `Static Method` to create instances of Customer Repo -> your method would still take in data, however your `customer repo` would now take in `instances of customers` as the parameter.

Reply:
Just reviewed and the default parameters look great!